composer/trainer/trainer.py

```diff
@@ -2018,6 +2018,9 @@ def _train_loop(self) -> None:
         finished_epoch_early = False
         last_wct = datetime.datetime.now()
 
+        if self.state.max_duration is None:
+            raise RuntimeError('max_duration must be specified when calling Trainer.fit()')
```

Comment:
Yea exactly - just fighting with the typing issue. I personally like keeping the user-facing error there in case something breaks down or the typing changes in the future. Will update the message though and add a comment